contracts/Paper.sol

```diff
@@ -51,13 +58,43 @@ contract Paper {
         require(block.timestamp <= deadlineDate, "The deadline date has passed.");
         _;
     }
+
+    modifier primaryCheckedHasPassed() {
+        require(primaryChecked == true);
+        _;
+    }
+
+    function primaryChecking() public onlyOwner() {
+        primaryChecked = true;
+    }
     
-    function addFeedback(string memory _feedback) public canBeReview() {
-        feedbacks.push(FeedBack(_feedback, msg.sender));
+    function addFeedback(string memory _feedback) public  primaryCheckedHasPassed() canBeReview() {
+        pendingFeedbacks.push(FeedBack(_feedback, msg.sender));
     }
 
+    function validateFeedback(string memory _validatedFeedback) public  primaryCheckedHasPassed() onlyOwner() {
+        for (uint i = 0; i < pendingFeedbacks.length; i++) {
+            if (keccak256(abi.encodePacked(pendingFeedbacks[i].feedback))
+            == keccak256(abi.encodePacked(_validatedFeedback))) {
+                feedbacks.push(FeedBack(pendingFeedbacks[i].feedback, pendingFeedbacks[i].reviewer));
+                delete pendingFeedbacks[i];
+            }
+        }
+        revert("No such feedback !");
+    }
```

Comment:
It is far more efficient to only give the index of the feedback in the pending feedback array. Also, you should store a boolean in the structure indicating if the feedback is pending or not, rather than moving it between arrays to save gas.

```suggestion
    function validateFeedback(uint _feedbackIndex) public  primaryCheckedHasPassed() onlyOwner() {
        feedbacks[_feedbackIndex].pending = false;
    }
```